src/crimson/common/interruptible_future.h

```diff
@@ -117,14 +117,18 @@ auto call_with_interruption_impl(
   // need to be interrupted, return an interruption; otherwise, restore the
   // global "interrupt_cond" with the interruption condition, and go ahead
   // executing the Func.
-  if (!interrupt_cond<InterruptCond> && interrupt_condition) {
+  if (interrupt_condition) {
```

Comment:
do you mean the continuation starts to run before the global "interrupt_cond" is not clear?